src/plots/Histogram.tsx

```diff
@@ -177,31 +170,153 @@ export default function Histogram({
           textposition: showBarValues ? 'auto' : undefined,
           marker: {
             ...(series.color ? { color: series.color } : {}),
-            opacity: perBarOpacity,
           },
           offset: 0,
           width: binWidths,
+          selected: {
+            marker: {
+              opacity: 1,
+            },
+          },
+          unselected: {
+            // switch off fading of unselected bars while selecting
+            marker: {
+              opacity: 1,
+            },
+          },
         };
       }),
     [data, orientation, calculatedBarOpacity, selectedRange]
   );
 
-  const handleSelectedRange = useCallback(
+  /**
+   * calculate midpoints of a unique set of bins
+   */
+  const binSummaries: BinSummary[] = useMemo(() => {
+    const allBins: HistogramBin[] = data.series
+      .map((series) => [series.bins])
+      .flat(2);
+
+    const seenLabels: string[] = [];
+    const uniqueBins = allBins.filter((bin) => {
+      const seenBefore = seenLabels.indexOf(bin.binLabel);
+      seenLabels.push(bin.binLabel);
+      return seenBefore < 0;
```

Comment:
I think this could be more efficient? Maybe doesn't matter though, even with lots of bins

```suggestion
      if (seenBefore < 0) {
        seenLabels.push(bin.binLabel);
        return true;
      }
      return false;
```

Reply:
Definitely more efficient - thanks!